src/solvers/smt2/smt2_dec.cpp

```diff
@@ -216,14 +216,11 @@ decision_proceduret::resultt smt2_dect::read_result(std::istream &in)
     }
   }
 
-  for(identifier_mapt::iterator
-      it=identifier_map.begin();
-      it!=identifier_map.end();
-      it++)
+  for(auto &it : identifier_map)
```

Comment:
Please avoid using `it` here as it isn't an iterator anymore.

Reply:
now "assignment"